### DISCUSSION TITLE
[Resolved] SoapySDR + RTLSDR-Airband on machine startup

### DISCUSSION THREAD

**Discussion by JackStromberg** [rtl-airband/RTLSDR-Airband#265] [General]
I noticed if you trigger a reboot of the machine itself, the initial start of SoapySDR to read streams may timeout and fail.  I've tried change After=network.target to After=network.target sdrplay.service in init.d, but still see an issue where you will experience a timeout: https://github.com/szpajder/RTLSDR-Airband/blob/master/input-soapysdr.cpp#L300  If I trigger a restart of the service, it'll immediately begin working again.

Taking a look at the Daemon logs, I don't see "Failed to activate stream for SoapySDR device", so I'm wondering if this is as simple as adding a delay between these two?  Not sure what happens during SoapySDRDevice_readStream, so kinda a shot in the dark on this.

```
pi@raspberrypi:~ $ sudo service rtl_airband status
● rtl_airband.service - SDR AM/NFM demodulator
   Loaded: loaded (/etc/systemd/system/rtl_airband.service; enabled; vendor preset: enabled)
   Active: inactive (dead) since Sat 2021-07-31 18:32:36 CDT; 20s ago
     Docs: https://github.com/szpajder/RTLSDR-Airband/wiki
  Process: 439 ExecStart=/usr/local/bin/rtl_airband -Fe (code=exited, status=0/SUCCESS)
 Main PID: 439 (code=exited, status=0/SUCCESS)

Jul 31 18:32:35 raspberrypi rtl_airband[439]: [496]: sdrplay_api_device: heartBeatThread: Exit
Jul 31 18:32:36 raspberrypi rtl_airband[439]: [494]: sdrplay_api_device: strmHandlerThread: Exit
Jul 31 18:32:36 raspberrypi rtl_airband[439]: [439]: sdrplay_api_Close
**Jul 31 18:32:36 raspberrypi rtl_airband[439]: SoapySDR device 'driver=sdrplay,hardware=RSPduo,mode=MA': readStream failed: TIMEOUT, disabl**
**Jul 31 18:32:36 raspberrypi rtl_airband[439]: Disabling device outputClosing connection** to xxxxxxxxxxxxxxxx
Jul 31 18:32:36 raspberrypi rtl_airband[439]: All receivers failed, exiting
Jul 31 18:32:36 raspberrypi rtl_airband[439]: Cleaning up
Jul 31 18:32:36 raspberrypi rtl_airband[439]: Input threads closed
Jul 31 18:32:36 raspberrypi rtl_airband[439]: Disabling device output
Jul 31 18:32:36 raspberrypi systemd[1]: rtl_airband.service: Succeeded.

```

**Comment by szpajder** [comment]
This is fixed in unstable branch (https://github.com/szpajder/RTLSDR-Airband/tree/unstable). `readStream` errors are no longer treated as fatal.

The fix will come out in the next stable release. Meanwhile you can use unstable branch, it's actually pretty stable :-) Note that the build instructions in the Wiki no longer apply to this branch - new instructions are in README.md. The Wiki will be updated when the new stable release is out.

**Reply by JackStromberg** [reply]
Excellent, thank you!  For reference: https://github.com/szpajder/RTLSDR-Airband/commit/de4dd4b3d52a4e21752c4b1dacff39ac5daaa5a7
